cpp/arcticdb/entity/protobuf_mappings.cpp

```diff
@@ -0,0 +1,140 @@
+/* Copyright 2023 Man Group Operations Limited
+ *
+ * Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+ *
+ * As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+ */
+#include <arcticdb/entity/protobufs.hpp>
+#include <arcticdb/entity/atom_key.hpp>
+#include <arcticdb/util/variant.hpp>
+#include <arcticdb/entity/protobuf_mappings.hpp>
+#include <arcticdb/entity/stream_descriptor.hpp>
+#include <arcticdb/entity/timeseries_descriptor.hpp>
+
+namespace arcticdb {
+
+using namespace arcticdb::entity;
+
+
+inline arcticdb::proto::descriptors::SortedValue sorted_value_to_proto(SortedValue sorted) {
+    switch (sorted) {
+    case SortedValue::UNSORTED:
+        return arcticdb::proto::descriptors::SortedValue::UNSORTED;
+    case SortedValue::DESCENDING:
+        return arcticdb::proto::descriptors::SortedValue::DESCENDING;
+    case SortedValue::ASCENDING:
+        return arcticdb::proto::descriptors::SortedValue::ASCENDING;
+    default:
+        return arcticdb::proto::descriptors::SortedValue::UNKNOWN;
+    }
+}
+
+inline SortedValue sorted_value_from_proto(arcticdb::proto::descriptors::SortedValue sorted_proto) {
+    switch (sorted_proto) {
+    case arcticdb::proto::descriptors::SortedValue::UNSORTED:
+        return SortedValue::UNSORTED;
+    case arcticdb::proto::descriptors::SortedValue::DESCENDING:
+        return SortedValue::DESCENDING;
+    case arcticdb::proto::descriptors::SortedValue::ASCENDING:
+        return SortedValue::ASCENDING;
+    default:
+        return SortedValue::UNKNOWN;
+    }
+}
+
+[[nodiscard]] arcticdb::proto::descriptors::IndexDescriptor index_descriptor_to_proto(const IndexDescriptorImpl& index_descriptor) {
+    arcticdb::proto::descriptors::IndexDescriptor proto;
+    proto.set_kind(static_cast<arcticdb::proto::descriptors::IndexDescriptor_Type>(index_descriptor.type_));
```

Comment:
TIME_SYMBOL is for the multi-compaction, part 2 of which is still on a branch. It's not in yet because I'm not 100% it's definitely going to work like it is on the branch (it most likely is, I just want to revisit with a fresh and critical eye).